pkg/tasks/prometheus.go

```diff
@@ -38,6 +39,26 @@ func NewPrometheusTask(client *client.Client, factory *manifests.Factory, config
 }
 
 func (t *PrometheusTask) Run(ctx context.Context) error {
+
+	errs := []error{}
+
+	err := t.create(ctx)
+	if err != nil {
+		klog.V(4).ErrorS(err, "updation of prometheus failed")
+		errs = append(errs, err)
+	}
+
+	// NOTE: the validation task is run even if creation fails so that existing
+	// deployment is validated.
+	validation := NewPrometheusValidationTask(t.client, t.factory)
+	if err = validation.Run(ctx); err != nil {
+		errs = append(errs, err)
+	}
```

Comment:
Since we create an `Aggregate` from this list, here is no need to check for nil, `NewAggregate` will do that.